include/dxc/Support/WinAdapter.h

```diff
@@ -1039,8 +1040,62 @@ class CHandle {
   HANDLE m_h;
 };
 
+
+/////////////////////////////////////////////////////////////////////////////
+// CComBSTR
+
+class CComBSTR
+{
+public:
+    BSTR m_str;
+    bool m_needFree;
+    CComBSTR() : m_str(nullptr), m_needFree(false) {};
+    CComBSTR(_In_ int nSize, LPCWSTR sz);
+    ~CComBSTR() throw() {
+    if (m_needFree)
+      SysFreeString(m_str);
+    }
+
+    operator BSTR() const throw()
+    {
+        return m_str;
+    }
+
+    bool operator==(_In_ const CComBSTR& bstrSrc) const throw();
+
+    BSTR* operator&() throw()
+    {
+        return &m_str;
+    }
+
+    BSTR Detach() throw()
+    {
+        BSTR s = m_str;
+        m_str = NULL;
```

Comment:
This does not reset `m_needFree`.  However, I don't see why we need `m_needFree` in the first place - see earlier comment.